### PR TITLE
Handle cart management of E-Shop page

### PR DESCRIPTION
**Pull request by st-lanphan**
- https://www.figma.com/file/CUQiZbPKOlfjHZn9hBqrPh/Fashion?node-id=0%3A1
image
> Cart management:
image

### CODE REVIEW

File: javaScript/cart.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+function getData() {
```
Comment (st-vantran) [inline javaScript/cart.js:1]: define ở file base.js.
Dùng listkey

File: javaScript/cart.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  var carts = window.localStorage.getItem("cart");
+  return JSON.parse(carts);
+}
+function renderCart() {
+  var cart = getData();
+  if (cart.length > 0) {
+    var containerCart = document.querySelector(".container");
+    var ulContainers = `<ul class="listCart"></ul>`;
+    containerCart.innerHTML += ulContainers;
```
Comment (st-vantran) [inline javaScript/cart.js:10]: Sao em không tạo sẵn thẻ `<ul>` ở file html nhỉ?

File: javaScript/cart.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    <a href="./fashion-home-1.html" class="btn btn-primary">Go Back</a>`;
+    totalCart.innerHTML = htmlGoBack;
+    removeProduct();
+  }
+}
+function handleListenerQuantity(id) {
+  var cartItem = getData();
+  cartItem.forEach(function(product) {
+    var htmlCartItemReduce = document.getElementById(`reduce-${product.id}`);
+    console.log(htmlCartItemReduce);
```
Comment (st-vantran) [inline javaScript/cart.js:72]: Xóa log

File: javaScript/index.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    return productElement;
+  });
+  productListElements.forEach(function (element) {
+    productElements.forEach(function (productElement) {
+      element.innerHTML += productElement;
+    });
+  });
+}
+
+function handleListenerButton(id) {
```
Comment (st-vantran) [inline javaScript/index.js:83]: id ở đây làm gì em nhỉ?

Comment (st-lanphan) [reply javaScript/index.js:83]: done

File: javaScript/cart.js (hunk 3 above)
Comment (st-lanphan) [reply javaScript/cart.js:72]: done

File: javaScript/cart.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  }
+  if (cart.length == 0) {
+    var totalCart = document.querySelector(".js-total");
+    var htmlGoBack = `<p>Bạn chưa thêm sản phẩm nào</p>
+    <a href="./fashion-home-1.html" class="btn btn-primary">Go Back</a>`;
+    totalCart.innerHTML = htmlGoBack;
+    removeProduct();
+  }
+}
+function handleListenerQuantity(id) {
```
[the remark this answers is not in the document]
Comment (st-lanphan) [reply javaScript/cart.js:68]: done

File: javaScript/base.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+var listKey = {
+  productList: "products",
+  cartList: "cart",
+}
+function setData(listKey,data) {
+  return window.localStorage.setItem(listKey, JSON.stringify(data));
+}
```
[the remark this answers is not in the document]
Comment (st-lanphan) [reply javaScript/base.js:7]: done

File: javaScript/cart.js (hunk 1 above)
Comment (st-lanphan) [reply javaScript/cart.js:1]: done

File: javaScript/cart.js (hunk 2 above)
Comment (st-lanphan) [reply javaScript/cart.js]: done

**Comment by st-linhnguyen** [comment]
Code is still not optimized. Please fix after convert to ts. I will merge this PR for temporary.

**Pull request merged**